### DISCUSSION TITLE
Can you make callouts collapsable by default?

### DISCUSSION THREAD

**Discussion by malcolmbarrett** [quarto-dev/quarto-cli#2869] [Q&A] [answered]
You can make callout blocks collapsable by adding `collapse="true"` as in https://quarto.org/docs/authoring/callouts.html#markdown-syntax.

Can you make this the default for all callout blocks in a project, e.g. via the project YAML?

**Comment by shafayetShafee** [comment, marked as answer]
I am not sure there's a straightforward YAML option to do this, but I can offer a solution using Lua filter.

**_quarto.yml**

~~~
project:
  type: book
  
book:
  title: "Quarto Book"
  chapters:
    - index.qmd
    - callouts.qmd
  
format: html
~~~

**callouts.qmd**

~~~
---
filters: 
 - callouts.lua
---

# Callout Blocks

:::{.callout-note}
Note that there are five types of callouts, including:
`note`, `warning`, `important`, `tip`, and `caution`.
:::

:::{.callout-tip}
## Tip With Caption

This is an example of a callout with a caption.
:::

:::{.callout-caution}
## Expand To Learn About Collapse

This is an example of a 'folded' caution callout that can be expanded by the user. You can use `collapse="true"` to collapse it by default or `collapse="false"` to make a collapsible callout that is expanded by default.
:::

:::{.callout-important}

This is important
:::


:::{.callout-warning icon='false'}

WARNINGS !!!!
:::
~~~

**callouts.lua**

~~~
function Div(el)
  if quarto.doc.isFormat("html") then
    callouts = {'callout-tip', 'callout-note', 'callout-warning', 
              'callout-caution', 'callout-important'}
    for key, val in pairs(callouts) do
      if el.classes:includes(val) then
        el.attributes["collapse"] = 'true'
        return el
      end
    end
  end
end 

~~~

<hr>

![image](https://user-images.githubusercontent.com/48413872/195999331-b4323d20-3179-4d91-9cd3-69c5c2053e0e.png)

<hr>

But I don't know how to add this lua filter to the project YAML file so that it applies all the `qmd` files.

**Reply by malcolmbarrett** [reply]
This worked great, thank you! All I needed to do was add the `filters` yaml to the top level of `_quarto.yml`

For anyone that's interested, here's what we did in practice: https://github.com/malcolmbarrett/causal-inference-in-R/pull/128/files#diff-469a07f57393912a9c17c3ea50ceebe991e32b008beb167a200b17e91fdac6d2

**Comment by cderv** [comment]
Just note that since Quarto 1.3, there are custom AST nodes, and `Callout` is one of them. 

here is an example of the Lua filter to use to achieve this 
- https://examples.quarto.pub/collapse-callout/

**Reply by haxfn** [reply]
ast seems awesome, i have zero skills in JS and zero hackability on web tech

**Reply by mcanouil** [reply]
@haxfn look for existing extensions then: https://m.canouil.dev/quarto-extensions/

This is available as one.

**Reply by haxfn** [reply]
Oh, I was able to use the lua code in the link shared by cderv. 

I was just wondering if ast would make hacking around quarto with treating qmd files as structured data to be processed in code. I guess ast does indeed allow that?
